### PR TITLE
#1-4 Task. Sergey Starushkin

### PR DESCRIPTION
**Pull request by Sergey-Starushkin**

### CODE REVIEW

File: task/01-strings-tasks.js [hunk 1]
```diff
@@ -55,7 +55,7 @@ function getStringLength(value) {
  *   'Chuck','Norris'  => 'Hello, Chuck Norris!'
  */
 function getStringFromTemplate(firstName, lastName) {
-    throw new Error('Not implemented');
+    return 'Hello, '+ firstName+' '+lastName+'!'
```
Comment (FedorovD) [inline task/01-strings-tasks.js:58]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

File: task/01-strings-tasks.js [hunk 2]
```diff
@@ -69,7 +69,8 @@ function getStringFromTemplate(firstName, lastName) {
  *   'Hello, Chuck Norris!' => 'Chuck Norris'
  */
 function extractNameFromTemplate(value) {
-    throw new Error('Not implemented');
+    let value2 = value.replace("Hello, ",'')
+    return value2.replace("!",'')
```
Comment (FedorovD) [inline task/01-strings-tasks.js:73]: redundant variable declarations, you just can do `return value.replace(...).replace(...)`

File: task/01-strings-tasks.js [hunk 3]
```diff
@@ -99,7 +100,7 @@ function getFirstChar(value) {
  *   '\tHello, World! ' => 'Hello, World!'
  */
 function removeLeadingAndTrailingWhitespaces(value) {
-    throw new Error('Not implemented');
+    return  value.trim()
```
Comment (FedorovD) [inline task/01-strings-tasks.js:103]: double space

File: task/01-strings-tasks.js [hunk 4]
```diff
@@ -130,7 +131,7 @@ function repeatString(value, count) {
  *   'ABABAB','BA' => 'ABAB'
  */
 function removeFirstOccurrences(str, value) {
-    throw new Error('Not implemented');
+    return  str.replace(value,'')
```
Comment (FedorovD) [inline task/01-strings-tasks.js:134]: double space

File: task/02-numbers-tasks.js [hunk 5]
```diff
@@ -221,7 +237,8 @@ function isPrime(n) {
  *   toNumber(new Number(42), 0) => 42
  */
 function toNumber(value, def) {
-    throw new Error('Not implemented');
+    if (parseInt(value,10)) { return parseInt(value,10); }
```
Comment (FedorovD) [inline task/02-numbers-tasks.js:240]: not correct if value = '0',  need a NaN check

**Comment by FedorovD** [comment]
lots of formatting issues:
lots of `;` and ` ` missing;
`var` is deprecated. It has no sense to use for browsers code even currently;
**missing travis link;**

image

Please clear PR code not related to Task#1-4 and click "ready for review" when it will be finished. I think the same issue with the other tasks (5-11), please fix them also. Thank you!

**Comment by vramaniuk** [comment]
you should have provided this direct link to your task success travis build
https://travis-ci.org/github/Sergey-Starushkin/js-assignments/builds/712292273

please, change your code according previous notes

and revert changes in files not related to the current task1-4 

please, see https://stackoverflow.com/questions/7147270/hard-reset-of-a-single-file

I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

**Comment by Sergey-Starushkin** [comment]
I checked several times, I have only the necessary files updated

**Comment by Sergey-Starushkin** [comment]
https://travis-ci.org/github/Sergey-Starushkin/js-assignments/builds/722844026

**Comment by vramaniuk** [comment]
We have 23 files changed. 
It will completely impossible in real project to find exactly yours changes, needed at that time.


You should check your history and revert dirty commits. 
Selection_012

I suppose, it can be be138aa5f1dc615d08d8fdcf1312f25b1d299fe0 and a89848f719562eff5323889e5f8aabc2c390ccc9

Invesitgate this https://github.com/AisBrestEDU/js-assignments/pull/645
He has 6 files changed (exacltly that needed) and the same version of ours repo as you have
Selection_013
https://github.com/s-samusik/js-assignments/commits/master

**Comment by vramaniuk** [comment]
Please, check you readme.md 
Compare with it
Selection_014

**Comment by vramaniuk** [comment]
And do not forget to implement FedorovD  notes

**Comment by vramaniuk** [comment]
I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

**Comment by Sergey-Starushkin** [comment]
I apologize for the error in the tests. I'll fix it soon.

**Pull request closed without merging**